charts/hub/portworx/templates/aws-prereqs/px-aws-prereqs-cm.yaml

```diff
@@ -0,0 +1,207 @@
+      NODE_REG=$(oc get node -l 'node-role.kubernetes.io/worker=' -o=jsonpath='{.items[0].metadata.name}')
+      AWS_REGION=$(oc get node $NODE_REG -o jsonpath='{.metadata.labels.topology\.kubernetes\.io/region}')
+    cat << EOF >> /pattern-home/credentials/.aws/credentials
+    [default]
+    aws_access_key_id = $AWS_ACCESS_KEY
+    aws_secret_access_key = $AWS_SECRET_KEY
+    EOF
+
+    cat << EOF >> /pattern-home/credentials/.aws/config
+    [default]
+    region = $AWS_REGION
+    output = json
+    EOF
+
+    fi
+
+
+    # Ensure we have AWS credentials to use
+    echo "Checking for AWS credentials."
+    aws sts get-caller-identity
+
+    if [ $? != 0 ];
+    then
+      echo "AWS credentials not found, exiting."
+      exit 1
+    else
+      echo "Found AWS credentials."
+    fi
+
+    # Ensure oc is functional and we can get node info from OCP cluster via serviceAccount
+    oc get node
+    if [ $? != 0 ];
+    then
+      echo "Cannot list nodes, please check rbac for portworx-prereq-sa serviceAccount."
+      exit 1
+    else
+      echo "Configuring for Portworx/Red Hat Multicloud GitOps Pattern."
+    fi
+    
+    # Ensure there are a minimum of three worker nodes in the cluster
+    NUM_WORKERS=$(oc get node -l 'node-role.kubernetes.io/worker=' -o name | wc -l)
+    if [ $NUM_WORKERS -lt 3 ];
+    then
+      echo "Only $NUM_WORKERS worker nodes detected - minimum required is three."
+      exit 1
+    else
+      echo "Found $NUM_WORKERS worker nodes - configuring..."
+    fi
+    
+    # Get the list of worker nodes in the cluster
+    WORKERS=$(oc get node -l 'node-role.kubernetes.io/worker=' -o name)
+
+    # Modify the SG created by OpenShift Installer to allow ports necessary for Portworx
+    for NODE in $WORKERS; do
+      unset AWS_INSTANCE
+      unset AWS_REGION
+      unset AWS_SG
+      AWS_INSTANCE=$(oc get $NODE -o jsonpath='{.spec.providerID}' | sed 's|.*/||')
+      AWS_REGION=$(oc get $NODE -o jsonpath='{.metadata.labels.topology\.kubernetes\.io/region}')
+      AWS_SG=$(aws ec2 describe-instances --instance-id $AWS_INSTANCE --region=$AWS_REGION --query "Reservations[].Instances[].SecurityGroups[].GroupId[]" --output text)
+      for GROUP in $AWS_SG; do
+        OCP_SG=$(aws ec2 describe-security-groups --group-ids $GROUP --query SecurityGroups[*].Description | grep -w "Created By OpenShift Installer" | wc -l)
+        if [ $OCP_SG -eq 1 ];
+        then
+          echo "Found OpenShift created security group assigned to $AWS_INSTANCE, opening Portworx ports."
+
+          # Check to see if security group already allows TCP 17001-17022
+          RULE_EXISTS_SG=$(aws ec2 describe-security-groups --region $AWS_REGION --filters Name=ip-permission.from-port,Values=17001 Name=ip-permission.to-port,Values=17022 Name=ip-permission.group-id,Values=$AWS_SG  --output text)

```

Comment:
These calls, could probably benefit from a function, since the task is being repeated a few times.